api/types/apitypes.go

```diff
@@ -471,37 +494,56 @@ func (s *Status) Height() uint32 {
 // if Status.height is the same as Status.dbHeight.
 func (s *Status) SetHeight(height uint32) {
 	s.Lock()
-	defer s.Unlock()
-	s.ready = height == s.dbHeight
+	s.ready = height == s.dbHeight && s.nodeConnections > 0
 	s.height = height
+	s.Unlock()
+}
+
+// DBHeight is the block most recently stored in the DB.
+func (s *Status) DBHeight() uint32 {
+	s.RLock()
+	defer s.RUnlock()
+	return s.dbHeight
 }
 
-// SetHeightAndConnections simultaneously sets the node height and node
-// connection count. Status.ready is set to true if the height and dbHeight are
-// the same.
-func (s *Status) SetHeightAndConnections(height uint32, conns int64) {
+// NodeConnections gets the number of node peer connections.
+func (s *Status) NodeConnections() int64 {
+	s.RLock()
+	defer s.RUnlock()
+	return s.nodeConnections
+}
+
+// SetConnections sets the node connection count.
+func (s *Status) SetConnections(conns int64) {
 	s.Lock()
-	defer s.Unlock()
 	s.nodeConnections = conns
-	s.ready = height == s.dbHeight
-	s.height = height
+	s.ready = s.ready && s.nodeConnections > 0
```

Comment:
Note that going back from 0 to N>0 connections does not automatically make it ready.  Restoring connections requires checking and setting node height, as done in `updateNodeConnections`.